File: jobhunter/backend/app/schemas/billing.py
from pydantic import BaseModel


class CheckoutRequest(BaseModel):
    tier: str


class CheckoutResponse(BaseModel):
    status: str
    url: str | None = None
    message: str | None = None


class PlanResponse(BaseModel):
    tier: str
    display_name: str
    price_monthly_cents: int
    description: str
    limits: dict[str, int]


class UpdatePlanRequest(BaseModel):
    plan_tier: str


class PortalResponse(BaseModel):
    url: str


class SubscriptionResponse(BaseModel):
    tier: str
    status: str
    current_period_end: str | None = None
    stripe_subscription_id: str | None = None

File: jobhunter/frontend/src/lib/api/billing.ts
import api from "./client";

export interface CheckoutResponse {
  status: string;
  url: string | null;
  message: string | null;
}

export interface SubscriptionResponse {
  tier: string;
  status: string;
  current_period_end: string | null;
  stripe_subscription_id: string | null;
}

export interface PortalResponse {
  url: string;
}

export async function createCheckout(tier: string): Promise<CheckoutResponse> {
  const { data } = await api.post<CheckoutResponse>(
    "/billing/create-checkout-session",
    { tier },
  );
  return data;
}

export async function createPortal(): Promise<PortalResponse> {
  const { data } = await api.get<PortalResponse>("/billing/portal");
  return data;
}

export async function getSubscription(): Promise<SubscriptionResponse> {
  const { data } = await api.get<SubscriptionResponse>(
    "/billing/subscription",
  );
  return data;
}
